src/mlpack/methods/ann/layer/max_pooling_impl.hpp

```diff
@@ -151,9 +151,15 @@ void MaxPoolingType<MatType>::Backward(
       this->inputDimensions[0], this->inputDimensions[1],
       channels * input.n_cols, false, true);
 
+  gTemp.zeros();
+
   // There's no version of UnpoolingOperation without pooling indices, because
   // if we call `Backward()`, we know for sure we are training.
-  UnpoolingOperation(mappedError, gTemp, poolingIndices);
+  #pragma omp parallel for
```

Comment:
Maybe we can push this into `UnpoolingOperation()`?

Reply:
I think it is giving some issues.

Reply:
Let me leave it as it is for now.
As it's not worth it to change.

Reply:
What are the issues that it is giving?  I think it is worth the time to debug.

Reply:
https://github.com/mlpack/mlpack/pull/3240/commits/1bf89fc9653e518a4d924e886c1151cc040b26ba

Reply:
Yes, I see that is the code you wrote, but what is the error it gives and what have you tried to debug it?

Reply:
I don't have any explanation for this as acc. to this code, nothing should have been changed in logic.

Mp its issue for private vs public variable in openMP.

Reply:
I haven't debugged it cause it will take some time.

Reply:
Ok, let's not worry about it then.